projects/spectator/src/lib/dom-selectors.ts

```diff
@@ -31,24 +31,27 @@ export const byText: DOMSelectorFactory<SelectorMatcherOptions> = (matcher, opti
 export const byTextContent = (matcher: Matcher, options: MandatorySelectorMatchingOptions): DOMSelector => {
   let textContentMatcher: Matcher;
   const normalizer: NormalizerFn = options?.normalizer || getDefaultNormalizer(options);
-  const getTextContent = (elem: HTMLElement): string => normalizer(elem.textContent ?? '');
+  const getTextContent = (elem: Element | null): string => normalizer(elem?.textContent ?? '');
 
-  if (typeof matcher === 'string') {
+  if (typeof matcher === 'string' || typeof matcher === 'number') {
     textContentMatcher = (_, elem) => {
       if (options?.exact === false) {
         return (
           getTextContent(elem)
             .toLowerCase()
-            .indexOf(matcher.toLowerCase()) >= 0
+            .indexOf(matcher.toString().toLowerCase()) >= 0
         );
       }
 
-      return getTextContent(elem) === matcher;
+      return getTextContent(elem) === matcher.toString();
     };
   } else if (matcher instanceof RegExp) {
     textContentMatcher = (_, elem) => matcher.test(getTextContent(elem));
-  } else {
+  } else if (typeof matcher === 'function') {
     textContentMatcher = (_, elem) => matcher(getTextContent(elem), elem);
+  } else {
+    // number Matcher not supported
```

Comment:
Outdated comment

Reply:
@benelliott I'm not familiar with this library.

Reply:
We could bump to a major release if you guys think this is a breaking change. @benelliott @johncrim

Reply:
It is technically a breaking change due to the change to `MatcherFunction` described above, but it's a fairly small breaking change at that.

Reply:
If it'll break someone app it should be a major release

Reply:
I agree, though it was not a major release for @testing-library/dom. I wish they hadn't made that change.

As long as it's not big deal to you, I think the major update is correct. This change does make a major version change to the @testing-library/dom dependency, which is part of the spectator API.